promql/engine_test.go

```diff
@@ -3501,3 +3501,65 @@ histogram {{sum:4 count:4 buckets:[2 2]}} {{sum:6 count:6 buckets:[3 3]}} {{sum:
 		},
 	})
 }
+
+func TestEvaluationWithDelayedNameRemovalDisabled(t *testing.T) {
+	opts := promql.EngineOpts{
+		Logger:                   nil,
+		Reg:                      nil,
+		EnableAtModifier:         true,
+		MaxSamples:               10000,
+		Timeout:                  10 * time.Second,
+		EnableDelayedNameRemoval: false,
```

Comment:
It was surprising to me that this is on in `NewTestEngine`.
Should we be testing with an experimental feature enabled by default?

Reply:
Should I change the default value to false instead

Reply:
I don't think it should change in this PR.  What do you think the default should be for tests?

Reply:
Tests run fairly quick we could run multiple times with some combination of flags maybe?